index.js

```diff
@@ -0,0 +1,150 @@
+const URL = 'https://ada-backtrek-api.herokuapp.com/trips/'
+
+
+const reportStatus = (message) => {
+  $('#status-message').html(message);
+};
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p><ul>`;
+  for (const field in errors){
+    for (const problem of errors[field]){
+      content += `<li>${field}: ${problem}</li>;`
+    }
+  }
+
+  content += "</ul>";
+  reportStatus(content);
+};
+
+const loadTrips = () => {
+  reportStatus('Loading trips ...');
+
+
+  const tripList = $('#trip-list');
+  tripList.empty();
+  $('#current-trips').removeClass('hidden')
+
+  axios.get(URL)
+  .then((response) => {
+    console.log(response);
+    reportStatus(`Successfully loaded`);
+    response.data.forEach((trip) => {
+
+      let id = trip.id
+
+      tripList.append(`<li class="trip" id="${id}">${trip.name}</li>`);
+
+
+    });
+    //only returns one element
+
+  })
+
+  .catch((error) => {
+    reportStatus(`Encountered an error while loading trips: ${error.message}`);
+    reportError(error.message, error.response.data.errors)
+    console.log(error);
+
+  });
+
+};
+
+const clickTrip = (trip) => {
+  console.log(trip)
+
+  $('#view-trip').removeClass('hidden')
+
+  axios.get(URL+trip)
+  .then((response) => {
+    console.log(response);
+    reportStatus(`Successfully loaded`);
+
+    console.log('appending to table body');
+
+    $('#tablebody').append(`<p>Name: ${response.data.name}</p>`);
+    $('#tablebody').append(`<p>Id: ${response.data.id}</p>`); $('#tablebody').append(`<p>Continent: ${response.data.continent}</p>`);
+     $('#tablebody').append(`<p>About: ${response.data.about}</p>`);
+    $('#tablebody').append(`<p>Category: </category>${response.data.category}</p>`);
```

Comment:
It might make sense to break the code to render trip details out into a separate function - as is, this function is doing two things (making the request, and rendering data). See also http://callbackhell.com/